lib/sprockets/npm.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+require 'json'
+
+module Sprockets
+  module Npm
+    # Internal: Override resolve_alternates to install package.json behavior.
+    #
+    # load_path    - String environment path
+    # logical_path - String path relative to base
+    #
+    # Returns candiate filenames.
+    def resolve_alternates(load_path, logical_path)
+      candidates, deps = super
+
+      # package.json can only be nested one level deep
+      if !logical_path.index('/'.freeze)
+        dirname = File.join(load_path, logical_path)
+
+        if directory?(dirname)
```

Comment:
I'm not sure if npm/bower modules are the right place to fix #169 since in those modules we are just extending the behavior defined in Resolve module. 

I would check if we can fix it in Resolve module since the bug seems to be happening with assets provided via gems.